docs/modeling/models.rst

```diff
@@ -424,240 +424,136 @@ the number of models, with one input per model::
-             [ 96., 100., 104., 108., 112., 116.]])
-      >>> y.shape
-      (5, 6)
+There are two ways to get around this. ``model_set_axis`` can be passed in
+when the model is evaluated::
+
+    >>> g(x, model_set_axis=1)
+    array([[1.        , 1.76499381],
+           [1.        , 1.76499381],
+           [1.        , 1.76499381]])
+
+Or when the model is initialized::
+
+    >>> g = models.Gaussian1D(amplitude=[[1, 2]], mean=[[0, 0]],
+    ...                       stddev=[[0.1, 0.2]], n_models=2,
+    ...                       model_set_axis=1)
+    >>> g(x)
+    array([[1.        , 1.76499381],
+           [1.        , 1.76499381],
+           [1.        , 1.76499381]])
+
+Note that in the latter case, the shape of the individual parameters has changed to 2D
+because now the parameters are defined along the 1st axis.
+
+The value of ``model_set_axis`` is either an integer number, representing the axis along which
+the different parameter sets and inputs are defined, or a boolean of value ``False``,
+in which case it indicates all model sets should use the same inputs on evaluation.
+For example, the above model has a value of 1 for ``model_set_axis``.
+If ``model_set_axis=False`` is passed the two models will be evaluated on the same input::
+
+    >>> g.model_set_axis
+    1
+    >>> result = g(x, model_set_axis=False)
+    >>> result
+    array([[[1.        , 0.60653066],
+            [2.        , 1.76499381]],
+    <BLANKLINE>
+           [[1.        , 0.60653066],
+            [2.        , 1.76499381]],
+    <BLANKLINE>
+           [[1.        , 0.60653066],
+            [2.        , 1.76499381]]])
+    >>> result[: , 0]
+    array([[1.        , 0.60653066],
+           [1.        , 0.60653066],
+           [1.        , 0.60653066]])
+    >>> result[: , 1]
+    array([[2.        , 1.76499381],
+           [2.        , 1.76499381],
+           [2.        , 1.76499381]])
+
+Currently model sets are most useful for fitting a set of **linear** models
+(:ref:`example <example-fitting-model-sets>`)
+allowing a large number of models of the same type to be fitted simultaneously
+(and independently from each other) to some large set of inputs, such as
+fitting a polynomial to the time response of each pixel in a data cube.
+This can greatly speed up the fitting process. The speed-up is due to solving
+the set of equations to find the exact solution. Nonlinear models, which require
+an iterative algorithm, cannot be currently fit using model sets. Model sets of nonlinear
+models can only be evaluated.
+
+
+.. _modeling-asdf:
+
+Model Serialization (Writing a Model to a File)
+===============================================
+
+Many models are serializable using the `ASDF`_
+format. This can be useful in many contexts, one of which is the implementation of a
+`Generalized World Coordinate System (GWCS)`_.
+Serializing a model to disk is possible by assigning the object to ``AsdfFile.tree``::
+
+.. doctest-requires:: asdf
```

Comment:
Somehow this text ends up visible in the HTML end product. That is not supposed to happen.